src/lib/utils/index.ts

```diff
@@ -159,12 +159,7 @@ export function selectByAddress<T>(
   map: Record<string, T>,
   address: string
 ): T | undefined {
-  const foundAddress = Object.keys(map).find(itemAddress => {
-    if (isSameAddress(itemAddress, address)) {
-      return true;
-    }
-  });
-  if (foundAddress) return map[foundAddress];
```

Comment:
The point of this function, and the other address helper functions, is that you can use them without worrying about the address casing. I understand why this is slow here, but maybe where the map is large we shouldn't be using this function rather than changing it?

Reply:
I wanted to ensure everything is fast, but I can see this function is used in a bunch of places where only a handful of tokens are in the map, so I've made a new fast version and am using that for the price/balance fetching stuff that goes over hundreds of tokens.